examples/stm32g4/src/bin/usb_serial.rs

```diff
@@ -0,0 +1,110 @@
+#![no_std]
+#![no_main]
+#![feature(type_alias_impl_trait)]
+
+use defmt::{panic, *};
+use embassy_executor::Spawner;
+use embassy_stm32::rcc::{ClockSrc, Pll, PllM, PllN, PllQ, PllR, PllSrc};
+use embassy_stm32::time::Hertz;
+use embassy_stm32::usb::{self, Driver, Instance};
+use embassy_stm32::{bind_interrupts, pac, peripherals, Config};
+use embassy_usb::class::cdc_acm::{CdcAcmClass, State};
+use embassy_usb::driver::EndpointError;
+use embassy_usb::Builder;
+use futures::future::join;
+use {defmt_rtt as _, panic_probe as _};
+
+bind_interrupts!(struct Irqs {
+    USB_LP => usb::InterruptHandler<peripherals::USB>;
+});
+
+#[embassy_executor::main]
+async fn main(_spawner: Spawner) {
+    let mut config = Config::default();
+
+    config.rcc.pll = Some(Pll {
+        source: PllSrc::HSE(Hertz(8000000)),
+        prediv_m: PllM::Div2,
+        mul_n: PllN::Mul72,
+        div_p: None,
+        // USB and CAN at 48 MHz
+        div_q: Some(PllQ::Div6),
+        // Main system clock at 144 MHz
+        div_r: Some(PllR::Div2),
+    });
+
+    config.rcc.mux = ClockSrc::PLL;
+
+    let p = embassy_stm32::init(config);
+    info!("Hello World!");
+
+    unsafe {
+        pac::RCC.ccipr().write(|w| w.set_clk48sel(0b10));
+    }
```

Comment:
Do you think we should add this HSI48 clk select in `config.rcc` as well.It seems to be the only way of having more than 144Mhz on the main CPU while using usb

![CleanShot 2023-06-14 at 14 21 53@2x](https://github.com/embassy-rs/embassy/assets/213276/6d551522-7419-4298-b7db-e572b4cd7429)

Reply:
I'll double check this. I'd like to have this example show how to use both the HSI48 and the PLL output. I'm not sure if there's any more setup that has to be done with the RCC to accomplish that or not. I know the HSI48 requires the clock recovery system to be setup to adjust it's frequency but I will take a look at it.

Reply:
Here's an example of using the CRS and the 48 MHz internal clock for USB: https://github.com/kevswims/stm32g4xx-hal/blob/develop/examples/usb_serial.rs

I'll see if I can get this added into the RCC as well though I don't think it needs to be part of the initial configuration since it is pretty independent of everything else.

Reply:
Similarly to the H5 config there is a boolean in the config to enable/disable the hsi48. But I agree the CK48 mux could be independent

Reply:
I'll add a flag to match that API then. That seems reasonable to me.

Reply:
From the generated STM32Cube code it just set the enable bit of the HSI48, no need to configure the other frequency recovery registers (I guess default parameters are fine). Basically the same as what is done in the H5

![CleanShot 2023-06-14 at 15 09 05@2x](https://github.com/embassy-rs/embassy/assets/213276/62ba58d4-2ff3-485c-a37c-f88c767c257b)

and 

![CleanShot 2023-06-14 at 15 10 38@2x](https://github.com/embassy-rs/embassy/assets/213276/234e2657-f49a-4e2a-a93b-8e10d7ae1c4d)

Reply:
The clock recovery is not required to turn it on but it is required to use it for USB applications since the oscillator itself can drift too much. The CRS uses info from the USB host to trim the frequency to be within the specs that are needed for USB operation.

Reply:
Ok I see, I was not aware of that part, good to know 😄